include/rive/message.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef _RIVE_MESSAGE_HPP_
+#define _RIVE_MESSAGE_HPP_
+
+#include "rive/math/vec2d.hpp"
+
+namespace rive {
+
+struct Message {
+    // TODO -- how to represent this?
+    
+    // JSON object?
```

Comment:
Maybe a `map` would work here? The lookup operators below would be efficient `O(1)` and values can be templatized.